torch/csrc/jit/script/compiler.cpp

```diff
@@ -1308,6 +1311,30 @@ struct to_ir {
     emitLoopCommon(stmt.range(), {}, {cond}, stmt.body(), {});
   }
 
+
+  // Currently we do not support assigning exceptions to variables,
+  // a = Exception("hi")
+  // raise a
+  //
+  // We don't validate that the name of the Exception
+  // raise AnyName(3)
+  //
+  // NYI: add exception logic to control-flow nodes
+  // if True:
+  //   a = 1
+  // else
+  //   raise Exception("Hi")
+  // print(a)
+  void emitRaise(const Raise& stmt) {
+    auto apply = Apply(stmt.expr());
```

Comment:
There is no guarantee that `stmt.expr()` is an apply node. I can write `raise 3 + 4`

Reply:
In python raise has to take in an exception. raise 3 + 4 doesn't work

Reply:
Good point, updated